articles/static-site-dynamic-nodejs-web-app/index.md

```diff
@@ -0,0 +1,146 @@
+Moving forward, we will discuss how to convert your static files to dynamic ones using the EJS templating engine and how to copy repeated code using partials and inject server-side variables to the front-end.~~
+
+### Templating Basics
+
+#### Installing EJS
+The first step to use EJS is to install it. A simple `npm install ejs --save` will do the trick. The --save parameter saves the module to the `package.json` file so anyone that clones the git repo (or otherwise downloads the site's files) can install all the required Node modules for the project (called dependencies) using the `npm install` command instead of having to type `npm install (module name)` for however many modules you need.
+
+### Converting Static Pages to EJS Files
+Next, you need to convert your static HTML files into dynamic EJS ones and set up your folder structure in the way EJS expects. In the root directory of your website, create a folder called views and inside that folder create two sub-folders called pages and partials. Move all your HTML files into the pages sub-folder and rename the .html file extensions to .ejs. Your folder structure should look similar to the picture below.
+
+![NodeJS File Structure](/engineering-education/static-site-dynamic-nodejs-web-app/nodejs-file-structure.png)
+
+### Reusing Code - Creating Your First EJS Partial
+When creating static sites, there's often code that you repeat on every page such as the head (where the meta tags are located), header, and footer sections. It's inconvenient to change them on every page (especially on larger sites) if alterations are needed, but if you use EJS partials then you won't have to. Editing one template (partial) file will update the code on every page that the file is included in.
+
+We'll take a typical part of a website to be templated, the header as an example. Create a new file called header.ejs in the partials folder. Copy and paste all the code between the `<header></header>` tags on one of your EJS pages into it.
+
+Finally, on all pages with a header delete the code between the `<header></header>` tags (the same code you copied to the header.ejs partial file) and replace it with `<% include('../partials/header') %>`. Now, you've created your first EJS partial. Repeat the process for any other repetitive pieces of code such as the head and footer sections.
+
+*Small Tip:* If you find it hard to differentiate between your pages and partials since they have the same .ejs file extension then it can be helpful to put an underscore _ in front of the names of partials so _ header.ejs. This is a naming convention that some developers use that can be helpful.
+
+### Rendering EJS Pages
+Now we get to the exciting part, making the server render the EJS pages and partials so you can see them on the front-end.
+
+First, we need to add the EJS Node module to our server. So, in the `server.js` file, add `const ejs = require('ejs');` near the beginning of the file with the other Node modules.
```

Comment:
'near the beginning' is nebulous... You should show exactly where the code goes within the context of server.js